tsfresh/feature_extraction/data.py

```diff
@@ -291,6 +295,124 @@ def __len__(self):
         return sum(grouped_df.ngroups for grouped_df in self.grouped_dict.values())
 
 
+def window(seq, window_width):
+    """
+    Returns a sliding window (of width n) over data from the iterable
+    s -> (s0,s1,...s[n-1]), (s1,s2,...,sn), ...
+
+    Taken from: https://stackoverflow.com/questions/6822725/rolling-or-sliding-window-iterator
+    """
+
+    # it = s0, s1, ...,
+    it = iter(seq)
+
+    # islice just takes the first n terms
+    # Therefore it iterates through the first n terms of it
+    # result = (it0, it1, ... itn)
+    result = tuple(itertools.islice(it, window_width))
+
+    # yield the first result
+    if len(result) == window_width:
+        yield result
+        # it = s[n+1], s[n+2], ...
+        # result = (it0, it1, ... itn) = (s0, s1, ..., sn)
+
+    # Iteratively drop the first term of result and append the first elem of 'it' to results
+    for elem in it:
+        result = result[1:] + (elem,)
+        yield result
+
+
+class RollingWideTsFrameAdapter(SliceableTsData):
+
+    def __init__(self, df, column_id, window_width, column_sort=None, value_columns=None):
+        """
+        Rolling window adapter for Pandas DataFrames in wide format, where multiple columns
+        contain different time series for the same id.
+
+        :param df: the data frame
+        :type df: pd.DataFrame
+
+        :param column_id: the name of the column containing time series group ids
+        :type column_id: str
+
+        :param window_width: the width of a single window in the rolling window
+        :type window_width: int
+
+        :param column_sort: the name of the column to sort on
+        :type column_sort: str|None
+
+        :param value_columns: list of column names to treat as time series values.
+            If `None`, all columns except `column_id` and `column_sort` will be used.
+        :type value_columns: list[str]|None
+        """
+
+        # pdb.set_trace()
+        _check_nan(df, column_id)
+
+        if value_columns is None:
+            value_columns = [col for col in df.columns if col not in [column_id, column_sort]]
```

Comment:
Since https://github.com/blue-yonder/tsfresh/pull/722 is merged (which you could of course not know), there is also a helper function for that available. Maybe you want to use this?

Reply:
Great stuff (I assume it's the `_get_value_columns` function)